File: app/src/main/java/arun/com/chromer/browsing/article/ImageViewActivity.java
/*
 * Chromer
 * Copyright (C) 2017 Arunkumar
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package arun.com.chromer.browsing.article;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;

import arun.com.chromer.R;
import arun.com.chromer.util.glide.GlideApp;

/**
 * Activity for viewing full size images from an article. Images are zoomable. You can pass in the
 * image url from an intent with .setData(url).
 */
public final class ImageViewActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.article_activity_image_view);
        ImageView imageView = findViewById(R.id.article_image_view);
        final String url = getIntent().getDataString();
        GlideApp.with(this)
                .load(url)
                .into(imageView);
    }
}

File: android-app/lynket/src/main/java/arun/com/chromer/browsing/article/ImageViewActivity.kt
/*
 *
 *  Lynket
 *
 *  Copyright (C) 2022 Arunkumar
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package arun.com.chromer.browsing.article

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import arun.com.chromer.R
import arun.com.chromer.util.glide.GlideApp
import kotlinx.android.synthetic.main.activity_image_view.*

/**
 * Activity for viewing full size images from an article. Images are zoomable. You can pass in the
 * image url from an intent with .setData(url).
 */
class ImageViewActivity : AppCompatActivity() {

  public override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_image_view)
    val url = intent.dataString
    GlideApp.with(this)
      .load(url)
      .into(imageView)
  }
}
